packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -506,6 +534,15 @@ class OffsetLayer extends ContainerLayer {
   /// pipeline.
   Offset offset;
 
+  @override
+  Object findRegion(Offset regionOffset, Type type) {
+    final Offset transformed = regionOffset - offset;
+    if (transformed.dx < 0.0 || transformed.dy < 0.0 ) {
```

Comment:
why are negative values not ok?

Reply:
Say you have a container layer with a list of offset layers as children.  How do you determine which layer you have "hit", since they layers themselves don't have bounds? 

Since we work backwards, anything with an offset less than zero seems like a miss

Reply:
You have to hit all of them in turn, I would assume. They're not clipped; they could paint outside their bounds. There's nothing about the origin that's especially special, especially if we're not doing something for the "other side" of the rectangle.

Reply:
Regarding the origin, by my own logic this should probably just check `y`.

Is it fairly common to paint outside of bounds?  We'd need some other metadata in the layer then so we could make a decision.

Reply:
I think I could grab the size of the AnnotatedRegion render object and use that as a sort of implicit clip just for the `find` api

Reply:
yeah passing the size in the same way the clip layers do it, and then limiting it in the region's findRegion, seems legit. Maybe pass through unclipped though if it doesn't match.

Reply:
pass through if it doesn't match the type or if it isn't within the clip?

Reply:
match the type.

Reply:
Done